extensions/oplogPopulator/modules/Connector.js

```diff
@@ -260,7 +274,19 @@ class Connector {
         this._buckets.delete(bucket);
         this._updateConnectorState(true);
         try {
-            await this.updatePipeline(doUpdate);
+            if (this._isPipelineImmutable && this._buckets.size > 1) {
+                this.logger.warning('Removing a bucket from an immutable pipeline', {
+                    method: 'Connector.removeBucket',
+                    connector: this._name,
+                    bucket,
+                });
+            } else if (this._isPipelineImmutable) {
+                // If the pipeline is immutable and only one bucket is left,
+                // we can destroy the connector, so it will be recreated with
+                // a new bucket later.
+                return this.destroy();
+            }
```

Comment:
The reason why we do this here is explained in the PR message

If we can safely delete a running connector, then we can decide to have this logic in the loop, to avoid deleting connectors possibly too frequently...